### ISSUE TITLE
Remove blocking due to commons-configuration reload behavior

### ISSUE THREAD

**Issue by twilmes** [JanusGraph/janusgraph#876]
I noticed some unexpected blocking during performance testing and traced it to Apache commons-configuration 1.10 making heavy use of synchronization. [Commons-configuration2](https://commons.apache.org/proper/commons-configuration/index.html) has been refactored to remove a number of these pain points so I think it would be a good idea to see if upgrading leads to performance improvements under heavy load.

Here is a sample stack demonstrating the issue:
```
gremlin-server-exec-13 State: BLOCKED CPU usage on sample: 96ms
org.apache.commons.configuration.AbstractFileConfiguration.reload(boolean) AbstractFileConfiguration.java:832
org.apache.commons.configuration.AbstractFileConfiguration.reload() AbstractFileConfiguration.java:827
org.apache.commons.configuration.AbstractFileConfiguration.containsKey(String) AbstractFileConfiguration.java:989
org.janusgraph.diskstorage.configuration.backend.CommonsConfiguration.get(String, Class) CommonsConfiguration.java:63
org.janusgraph.diskstorage.configuration.MixedConfiguration.has(ConfigOption, String[]) MixedConfiguration.java:43
org.janusgraph.diskstorage.configuration.MergedConfiguration.has(ConfigOption, String[]) MergedConfiguration.java:35
org.janusgraph.diskstorage.configuration.MergedConfiguration.get(ConfigOption, String[]) MergedConfiguration.java:43
org.janusgraph.diskstorage.util.StandardBaseTransactionConfig.getCustomOption(ConfigOption) StandardBaseTransactionConfig.java:87
org.janusgraph.graphdb.transaction.StandardTransactionBuilder$ImmutableTxCfg.getCustomOption(ConfigOption) StandardTransactionBuilder.java:558
org.janusgraph.diskstorage.cql.CQLTransaction.<init>(BaseTransactionConfig) CQLTransaction.java:37
org.janusgraph.diskstorage.cql.CQLStoreManager.beginTransaction(BaseTransactionConfig) CQLStoreManager.java:328
org.janusgraph.diskstorage.locking.consistentkey.ExpectedValueCheckingStoreManager.beginTransaction(BaseTransactionConfig) ExpectedValueCheckingStoreManager.java:86
org.janusgraph.diskstorage.locking.consistentkey.ExpectedValueCheckingStoreManager.beginTransaction(BaseTransactionConfig) ExpectedValueCheckingStoreManager.java:37
org.janusgraph.diskstorage.Backend.beginTransaction(TransactionConfiguration, KeyInformation$Retriever) Backend.java:529
org.janusgraph.graphdb.database.StandardJanusGraph.openBackendTransaction(StandardJanusGraphTx) StandardJanusGraph.java:348
org.janusgraph.graphdb.database.StandardJanusGraph.newTransaction(TransactionConfiguration) StandardJanusGraph.java:338
org.janusgraph.graphdb.transaction.StandardTransactionBuilder.start() StandardTransactionBuilder.java:239
org.janusgraph.graphdb.database.StandardJanusGraph.newThreadBoundTransaction() StandardJanusGraph.java:331
org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsGraph.startNewTx() JanusGraphBlueprintsGraph.java:82
org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsGraph.access$000(JanusGraphBlueprintsGraph) JanusGraphBlueprintsGraph.java:47
org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsGraph$GraphTransaction.doOpen() JanusGraphBlueprintsGraph.java:267
org.apache.tinkerpop.gremlin.structure.util.AbstractTransaction.open() AbstractTransaction.java:95
org.apache.tinkerpop.gremlin.structure.Transaction$READ_WRITE_BEHAVIOR$1.accept(Transaction) Transaction.java:210
org.apache.tinkerpop.gremlin.structure.Transaction$READ_WRITE_BEHAVIOR$1.accept(Object) Transaction.java:207
org.apache.tinkerpop.gremlin.structure.util.AbstractThreadLocalTransaction.doReadWrite() AbstractThreadLocalTransaction.java:92
org.apache.tinkerpop.gremlin.structure.util.AbstractTransaction.readWrite() AbstractTransaction.java:148
org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsGraph.getAutoStartTx() JanusGraphBlueprintsGraph.java:71
org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsGraph.addVertex(Object[]) JanusGraphBlueprintsGraph.java:129
org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsGraph.addVertex(Object[]) JanusGraphBlueprintsGraph.java:47
org.apache.tinkerpop.gremlin.process.traversal.step.map.AddVertexStartStep.processNextStart() AddVertexStartStep.java:85
org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext() AbstractStep.java:143
org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next() ExpandableStepIterator.java:50
org.apache.tinkerpop.gremlin.process.traversal.step.map.MapStep.processNextStart() MapStep.java:36
org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext() AbstractStep.java:143
org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next() ExpandableStepIterator.java:50
org.apache.tinkerpop.gremlin.process.traversal.step.map.MapStep.processNextStart() MapStep.java:36
org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext() AbstractStep.java:143
org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next() ExpandableStepIterator.java:50
org.apache.tinkerpop.gremlin.process.traversal.step.map.MapStep.processNextStart() MapStep.java:36
org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext() AbstractStep.java:143
org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next() ExpandableStepIterator.java:50
org.apache.tinkerpop.gremlin.process.traversal.step.map.MapStep.processNextStart() MapStep.java:36
org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext() AbstractStep.java:143
org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next() ExpandableStepIterator.java:50
org.apache.tinkerpop.gremlin.process.traversal.step.map.MapStep.processNextStart() MapStep.java:36
org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext() AbstractStep.java:143
org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next() ExpandableStepIterator.java:50
org.apache.tinkerpop.gremlin.process.traversal.step.map.MapStep.processNextStart() MapStep.java:36
org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext() AbstractStep.java:143
org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next() ExpandableStepIterator.java:50
org.apache.tinkerpop.gremlin.process.traversal.step.map.MapStep.processNextStart() MapStep.java:36
org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext() AbstractStep.java:143
org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next() ExpandableStepIterator.java:50
org.apache.tinkerpop.gremlin.process.traversal.step.map.MapStep.processNextStart() MapStep.java:36
org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext() AbstractStep.java:143
org.apache.tinkerpop.gremlin.process.traversal.step.util.ExpandableStepIterator.next() ExpandableStepIterator.java:50
org.apache.tinkerpop.gremlin.process.traversal.step.map.MapStep.processNextStart() MapStep.java:36
org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.hasNext() AbstractStep.java:143
org.apache.tinkerpop.gremlin.process.traversal.util.DefaultTraversal.hasNext() DefaultTraversal.java:192
org.apache.tinkerpop.gremlin.server.op.AbstractOpProcessor.handleIterator(Context, Iterator) AbstractOpProcessor.java:89
org.apache.tinkerpop.gremlin.server.op.AbstractEvalOpProcessor.lambda$evalOpInternal$5(RequestMessage, Context, boolean, Settings, Object) AbstractEvalOpProcessor.java:268
org.apache.tinkerpop.gremlin.server.op.AbstractEvalOpProcessor$$Lambda$239.accept(Object)
org.apache.tinkerpop.gremlin.groovy.engine.GremlinExecutor.lambda$eval$0(GremlinExecutor$LifeCycle, Bindings, String, String, CompletableFuture, long) GremlinExecutor.java:301
org.apache.tinkerpop.gremlin.groovy.engine.GremlinExecutor$$Lambda$157.call()
java.util.concurrent.FutureTask.run() FutureTask.java:266
java.util.concurrent.Executors$RunnableAdapter.call() Executors.java:511
java.util.concurrent.FutureTask.run() FutureTask.java:266
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor$Worker) ThreadPoolExecutor.java:1149
java.util.concurrent.ThreadPoolExecutor$Worker.run() ThreadPoolExecutor.java:624
java.lang.Thread.run() Thread.java:748
```

**Comment by mad** [comment]
It will be fixed after upgrade to TP 3.5 (https://issues.apache.org/jira/browse/TINKERPOP-2185)

**Comment by mad** [comment]
fixed by #2611
